packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -0,0 +1,35 @@
+import { createIntl, createIntlCache } from 'react-intl';
+import { messages } from '../localization';
+import { useMemo } from 'react';
+
+const DEFAULT_LOCALE = 'en-US';
+const cache = createIntlCache();
+
+/**
+ * The hook returns an instance of the Intl object with the specified locale
+ * and messages using imperative API.
+ * @see https://formatjs.io/docs/react-intl/api/#createintl
+ * @param {string} locale - The locale to use.
+ * @returns the `intl` object.
+ */
+export default function useImperativeIntl(locale) {
+  // NOTE: We need to remove the dash from the locale to match the messages file keys, ex: 'es-ES' -> 'esES'
```

Comment:
Why not just renaming the localization files?

Reply:
Do you mean es-ES.js for example, with dash?

Reply:
Done